sys/include/unaligned.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2019 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_unaligned unaligned access methods
+ * @ingroup     sys
+ * @brief       Provides functions for safe unaligned memory accesses
+ *
+ * This header provides functions to read values from pointers that are not
+ * necessarily aligned to the type's alignment requirements.
+ *
+ * E.g.,
+ *
+ *     uint16_t *foo = 0x123;
+ *     printf("%u\n", *foo);
+ *
+ * ... might cause an unaligned access, if uint16_t's is usually aligned ad
+ * 2-byte-boundaries, as foo has an odd address.
+ *
+ * The current implementation casts a pointer to a packed struct, which forces
+ * the compiler to deal with possibly unalignedness.  Idea taken from linux
+ * kernel sources.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Unaligned but safe memory access functions
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef UNALIGNED_H
+#define UNALIGNED_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Unaligned access helper struct (uint16_t version) */
+typedef struct {
+    uint16_t val;       /**< value */
+} __attribute__((packed)) uint16_una_t;
```

Comment:
Most of our own code puts the attribute directly after `struct`:

```
$ git grep "__attribute__((packed))" | wc -l
97
$ git grep "\(struct\|union\) __attribute__((packed))" | wc -l
91
$ git grep "__attribute__((packed))" | grep -v "\(struct\|union\) __attribute__((packed))"
cpu/esp32/vendor/esp-idf/include/esp32/esp_mesh.h:} __attribute__((packed)) mip_t;
cpu/esp32/vendor/esp-idf/include/esp32/esp_mesh.h:} __attribute__((packed)) mesh_opt_t;
cpu/esp32/vendor/esp-idf/include/esp32/esp_mesh_internal.h:} __attribute__((packed)) mesh_assoc_t;
cpu/esp8266/vendor/espressif/c_types.h:/* #define __packed        __attribute__((packed)) */
pkg/lwip/include/arch/cc.h:#define PACK_STRUCT_STRUCT      __attribute__((packed))
sys/include/net/gnrc/pktbuf.h: *          (defined with `__attribute__((packed))`) or enforce alignment in
```

Reply:
fixed